src/box/lua/mkversion.lua

```diff
@@ -25,16 +28,17 @@ local function mkversion_new(major, minor, patch)
     self.major = major
     self.minor = minor
     self.patch = patch
-    self.id = bit.bor(bit.lshift(bit.bor(bit.lshift(major, 8), minor), 8),
-                      patch)
+    self._id = bit.bor(bit.lshift(bit.bor(bit.lshift(major, 8), minor), 8),
+                       patch)
     return self
 end
 
 -- Parse string with version in format 'A.B.C' to version object.
 local function mkversion_from_string(version)
-    local major, minor, patch = version:match('^(%d+)%.(%d+)%.(%d+)$')
+    local major, minor, patch = version:match('^(%d+)%.(%d+)%.(%d+)[^%.]*$')
```

Comment:
Just comes into my head: it means that the following code that returns true now:

```lua
local v = require('version')

return v.fromstr('1.2.3.4') == v.fromstr('1.2.3.5')
```

...would return `false` if we dedice to support more components in a future.

And it means that we can't extend it due to compatibility concerns.

Reply:
Let's support full version specification right now? It shouldn't be difficult AFAIU so why put it off?

Reply:
> Let's support full version specification right now? It shouldn't be difficult AFAIU so why put it off?

But it will be independent from the C `version` module we have. Are we ok with that? 

Because I won't touch `src/version.c` module. It's used in replication, there're a lot of version dependent checks in `relay` and `applier`. If the way, we create `version_id` will be changed in C, then we'll have to introduce new flags in replication, which will say which version of the `version` module replica uses. I
don't wanna deal with that for now and see no reason for that actually (except of the code cleanliness). 

P.S. I'll completely get rid of the `mkversion`, we'll have two independent versions of the `version` module: public one (in the current patch), and C internal (`src/version`)

Reply:
> > Let's support full version specification right now? It shouldn't be difficult AFAIU so why put it off?
> 
> But it will be independent from the C `version` module we have. Are we ok with that?

I think it's okay as long as the Lua `version` module is compatible with it, i.e. it can parse strings generated with the C `version`. In the Lua module we could implement comparisons field-by-field, without using the version id. We don't need to deal with version labels (like alpha, beta, entrypoint) in replication or schema so the version representation would remain the same in C.

Reply:
> Let's support full version specification right now?

Done